rotary/rules/annotation.smk

```diff
@@ -309,16 +309,18 @@ if POLISH_WITH_SHORT_READS == True:
             "{sample}/logs/annotation/calculate_final_short_read_coverage.log"
         benchmark:
             "{sample}/benchmarks/annotation/calculate_final_short_read_coverage.txt"
+        params:
+            mem_per_thread = int(config.get("memory") / config.get("threads",1))
```

Comment:
Note that I generally changed how rules with samtools handle memory. Before, the mem per thread value (required by samtools) was set in `resources`. However, I realized that this could cause a bug with resource management, because to snakemake, this implies that the max. memory that rule will use is the GB per thread, which is incorrect. Now, I specify the global mem in `resources` and then calculate the mem per thread in `params`.